src/Config/Converters/McpOptionsConverterFactory.cs

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+
+namespace Azure.DataApiBuilder.Config.Converters;
+
+/// <summary>
+/// JSON converter factory for DmlToolsConfig that handles both boolean and object formats.
+/// </summary>
+internal class McpOptionsConverterFactory : JsonConverterFactory
+{
+    /// <inheritdoc/>
+    public override bool CanConvert(Type typeToConvert)
+    {
+        return typeToConvert.IsAssignableTo(typeof(DmlToolsConfig));
+    }
+
+    /// <inheritdoc/>
+    public override JsonConverter? CreateConverter(Type typeToConvert, JsonSerializerOptions options)
+    {
+        return new DmlToolsConfigConverter();
+    }
+
+    private class DmlToolsConfigConverter : JsonConverter<DmlToolsConfig>
+    {
+        /// <summary>
+        /// Reads DmlToolsConfig from JSON which can be either:
+        /// - A boolean: all tools are enabled/disabled
+        /// - An object: individual tool settings
+        /// </summary>
+        public override DmlToolsConfig? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType is JsonTokenType.Null)
+            {
+                return null;
+            }
+
+            // Handle boolean format: "dml-tools": true/false
+            if (reader.TokenType is JsonTokenType.True || reader.TokenType is JsonTokenType.False)
+            {
+                bool enabled = reader.GetBoolean();
+                return DmlToolsConfig.FromBoolean(enabled);
+            }
+
+            // Handle object format
+            if (reader.TokenType is JsonTokenType.StartObject)
+            {
+                bool? describeEntities = null;
+                bool? createRecord = null;
+                bool? readRecord = null;
+                bool? updateRecord = null;
+                bool? deleteRecord = null;
+                bool? executeRecord = null;
+
+                while (reader.Read())
+                {
+                    if (reader.TokenType is JsonTokenType.EndObject)
+                    {
+                        return new DmlToolsConfig
+                        {
+                            AllToolsEnabled = false, // Default when using object format
```

Comment:
default should be true. What if in object format, the value is : 

`"dml-tools" : { }`

Are all tools enabled or disabled in this case? 

Since this can be interpreted as absence of the sub-properties, and by default they are true. `AllToolsEnabled` should be true since we rely on this value in the function: `IsToolEnabled`